### PR TITLE
Run add_enr for each RFC test and remove pings

### PR DESCRIPTION
**Pull request by kdeme**
Add_enr is better specified for each ENR than what a ping does, it should be also faster for this test, which its purpose is not testing the pings.

### CODE REVIEW

**Comment by kdeme** [comment]
I think this is the better version compared to this fix https://github.com/ethereum/portal-hive/pull/93

We might want to add a test that specifically tests ENR adding over ping (discv5 handshakes).

**Comment by KolbyML** [comment]
I added Ping cause Mike made it so RFC only returned enrs to start the RFC search for "connected" nodes. There wasn't a clean was for us to "connect" a node without pinging it, I guess we could have sent a ping request in add_enr but I didn't think of that at the time. Later Mike made it so if there was 0 "connected" nodes then and only then we would use them as a starting point for RFC. So we should be able to remove the pings now. As that problem is resolved I believe.

**Review by KolbyML — APPROVED** [review]
If Trin passes 100/100 with this PR you can merge it. It should now if my memory serves me correctly.

**Comment by KolbyML** [comment]
In Trin ping also adds the enr which I guess is unlike fluffy which was stated in this review comment https://github.com/ethereum/portal-hive/pull/84#discussion_r1293199163 on that PR which added this code. Which is why add_enr was removed as it was assumed to be implicit.

I wonder if that behavoir should be up to implementors or spec-ed out

**Comment by KolbyML** [comment]
I just ran the tests locally with this PR and they all pass so this is good to merge

File: simulators/portal-interop/src/main.rs [hunk 1]
```diff
@@ -588,9 +588,12 @@ dyn_async! {
...
 
-        // send a ping so client A sees it as a seen/connected node
-        if let Err(err) = client_a.rpc.ping(target_enr.clone()).await {
-                panic!("Unable to receive pong info: {err:?}");
+        match HistoryNetworkApiClient::add_enr(&client_a.rpc, target_enr.clone()).await {
+            Ok(response) => match response {
+                true => (),
+                false => panic!("AddEnr expected to get true and instead got false")
+            },
+            Err(err) => panic!("{}", &err.to_string()),
```
Comment (carver) [inline simulators/portal-interop/src/main.rs:596]: *nit* interpolation simplifies, then some extra context in the error is nice:
```suggestion
            Err(err) => panic!("Failed while trying to add client B's ENR to client A: {err}"),
```

File: simulators/portal-interop/src/main.rs [hunk 2]
```diff
@@ -659,11 +662,6 @@ dyn_async! {
             Err(err) => panic!("{}", &err.to_string()),
         }
 
-        // send a ping so client A sees it as a seen/connected node
-        if let Err(err) = client_a.rpc.ping(target_enr.clone()).await {
```
Comment (carver) [inline simulators/portal-interop/src/main.rs:663]: Just breezing by without full context: why is this the only one of the three that doesn't also require an `add_enr`?

**Review by carver — APPROVED** [review]
> Add_enr is better specified for each ENR than what a ping does

Hmm, it seems that `portal_historyAddEnr` isn't any clearer about whether we should add the node in a connected state. (trin doesn't AFAICT) I don't have any objection to specifying that, but it seems like a good idea to add that to the [rpc description](https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/ethereum/portal-network-specs/assembled-spec/jsonrpc/openrpc.json&uiSchema%5BappBar%5D%5Bui:splitView%5D=false&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false).

LGTM

File: simulators/portal-interop/src/main.rs (hunk 2 above)
Comment (KolbyML) [reply simulators/portal-interop/src/main.rs:663]: Because this one already has add_enr it just also had ping

**Comment by carver** [comment]
> whether we should add the node in a connected state. (trin doesn't AFAICT)

Also, I'm seeing there's already a more subtle conversation going on in Discord about all this (ie~ differentiating between seen & connected). So feel free to ignore me, rather than repeat yourself here. 😛

**Comment by kdeme** [comment]
> > Add_enr is better specified for each ENR than what a ping does
> 
> Hmm, it seems that `portal_historyAddEnr` isn't any clearer about whether we should add the node in a connected state. (trin doesn't AFAICT) I don't have any objection to specifying that, but it seems like a good idea to add that to the [rpc description](https://playground.open-rpc.org/?schemaUrl=https://raw.githubusercontent.com/ethereum/portal-network-specs/assembled-spec/jsonrpc/openrpc.json&uiSchema%5BappBar%5D%5Bui:splitView%5D=false&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false).
> 
> LGTM

Yeah, that's a good point. In our implementation `portal_historyAddEnr` does immediately set the node as verified. As I use this call already in Fluffy CI tests to make those tests more deterministic.
It shouldn't be needed here in theory, at least if the implementation allows to message unverified nodes with other requests than ping. But it might be useful in other tests (where we want to see the node returned on a FindNodes call).

**Comment by kdeme** [comment]
I've tested this PR and also https://github.com/ethereum/portal-hive/pull/93, both work 100% for Trin.

I think I prefer this version, with an added specific that checks if an ENR got added on a `Ping`.

File: simulators/portal-interop/src/main.rs [hunk 3]
```diff
@@ -588,9 +588,12 @@ dyn_async! {
...
         };
 
-        // send a ping so client A sees it as a seen/connected node
-        if let Err(err) = client_a.rpc.ping(target_enr.clone()).await {
-                panic!("Unable to receive pong info: {err:?}");
+        match HistoryNetworkApiClient::add_enr(&client_a.rpc, target_enr.clone()).await {
+            Ok(response) => match response {
+                true => (),
+                false => panic!("AddEnr expected to get true and instead got false")
+            },
```
Comment (ogenev) [inline simulators/portal-interop/src/main.rs:595]:
```suggestion
            Ok(response) => if !response {
                panic!("AddEnr expected to get true and instead got false")
            },
```

**Review by ogenev — APPROVED** [review]

File: simulators/portal-interop/src/main.rs (hunk 2 above)
Comment (kdeme) [reply simulators/portal-interop/src/main.rs:663]: Yeah, the usage of add_enr / ping was inconsistent, and this one did both.

**Pull request merged**